src/store/userActions.js

```diff
@@ -0,0 +1,26 @@
+const USERNAME_KEY = 'username';
+
+export const setUsername = username => {
+    global.localStorage.setItem(USERNAME_KEY, username);
+
+    return {type: 'SET_NAME', name: username};
+};
+
+export const generateRandomUsername = () => {
+    const getRandomName = () => {
+        const randomUsernames = ['squirrel', 'cat', 'dog', 'horse', 'bird', 'hamster', 'snake', 'elephant', 'lion', 'panda'];
+
+        return randomUsernames[Math.floor(Math.random() * randomUsernames.length)];
+    };
+
+    return setUsername(getRandomName());
+};
+
+export const loadUsername = () => {
+    const username = global.localStorage.getItem(USERNAME_KEY);
```

Comment:
I don't know if this is pure or not, but I would put this check in app.jsx to prevent dispatching empty event. Any thoughts?